File: ConsoleFS/Demo3.fs
module Demo3

let add1 x = x + 1
let double x = x * 2
let square x = x * x

let add1DoubleAndSquare = add1 >> double >> square

let run () =
    let resultWithPipe = 5 |> add1 |> double |> square
    let result = add1DoubleAndSquare 5
    printfn "Result: %d" result

File: ConsoleCS/Demo3.cs
namespace ConsoleCS
{
    internal class Demo3
    {
        public static void Run()
        {
            Func<int,int> add1 = x => x + 1;
            Func<int,int> doubleIt = x => x * 2;
            Func<int,int> square = x => x * x;

            Func<int, int> add1DoubleAndSquare = x => square(doubleIt(add1(x)));
            int result = add1DoubleAndSquare(5);

        }
    }
}
